### PR TITLE
Add support for copying blobs, streaming upload of blobs

### PR DESCRIPTION
**Pull request by pwcsquared**
This PR adds a couple features that I needed for a recent project: copying an existing blob to a new location, and uploading a blob with a stream.

Blobs that are uploaded as streams will be created as as list of individual [blocks](https://learn.microsoft.com/en-us/rest/api/storageservices/operations-on-block-blobs). When the stream completes, the list of block IDs are committed to the blob. Because it may not be possible to know the content-type ahead of time for a stream, it will fall back to `application/octet-stream` as a default type (to match the default type that [Azure would set](https://learn.microsoft.com/en-us/rest/api/storageservices/put-blob?tabs=azure-ad#request-headers-all-blob-types)).

This PR also includes a few peripheral changes/additions in support of these features:
- bumped the `x-ms-version` to a more recent date
- Added the `Azurex.Blob.Block` module for working with block blobs
- added the test support module `AzuriteSetup` for setting test configuration and creating integration test containers and test blobs automatically when mix test is run with the `--include integration` option (instead of having to do it manually)

Open to any feedback, and thanks for all your work on this project!

### CODE REVIEW

File: lib/azurex/blob/block.ex [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+    )
+    |> HTTPoison.request()
+    |> case do
+      {:ok, %HTTPoison.Response{status_code: 201}} -> :ok
+      {:ok, err} -> {:error, err}
+      {:error, err} -> {:error, err}
+    end
+  end
+
+  defp build_block_id do
```
Comment (jakobht) [inline lib/azurex/blob/block.ex:98]: I think this function could be replaced by
```elixir
:rand.bytes(64) |> Base.encode64()
```
So I think, keep the function but replace the implementation with the above

File: lib/azurex/blob.ex [hunk 2]
```diff
@@ -29,11 +30,25 @@ defmodule Azurex.Blob do
...
+  This argument can be either a valid string, or `nil`. A `content_type`
+  argument of `nil` will result in the blob being assigned the default content
+  type `"application/octet-stream"`.
+
   ## Examples
 
       iex> put_blob("filename.txt", "file contents", "text/plain")
       :ok
 
+      iex> put_blob("filename.txt", {:stream, bitstream}, nil)
```
Comment (jakobht) [inline lib/azurex/blob.ex]: Could you define an example bitstream, so it' s a bit easier to understand the example, something like this:
```elixir
iex> bitstream = <something> 
...> put_blob("filename.txt", {:stream, bitstream}, nil)
:ok
```
where you replace `<something>` with an example

**Review by jakobht — CHANGES_REQUESTED** [review]
Thanks a lot for this! This is a lot of work. 

I added a few small comments, and then we can merge.

File: lib/azurex/blob/block.ex (hunk 1 above)
Comment (pwcsquared) [reply lib/azurex/blob/block.ex:98]: I like this idea, but it looks like `:rand.bytes/1` is not available until OTP 24.0, which is only supported by Elixir versions 1.11.4 and up. There is also `:crypto.strong_rand_bytes/1`, but both of these functions create random binaries, which may not be valid strings.

I did notice that I'm padding the generated ID with too many zeroes so it's ending up twice as long as I intended. I'll tighten that up a bit!

**Review by jakobht — APPROVED** [review]

**Pull request merged**